CDCgov/prime-reportstream
# Fetch Refactor: Establish ErrorBoundary and Suspense use patterns

This PR iterates on Doug's PR that refactored `ValueSetsIndex` and `ValueSetsDetail` to use the new react-query implementation. 

This creates a custom error boundary and adds `RSErrorBoundary` and `Suspense` wrappers for components making network calls. This boundary runs on RSNetworkErrors, and currently displays a generic UI.

This PR **does not** handle the implementation of the new ErrorBoundary across the app. That should be a follow-up PR.

Test Steps:
1. Pull this branch and run `yarn`
2. Load up the value-sets page from the Admin dropdown
3. Everything should load normally: with a spinner.
4. Go to `UseValueSets.ts` and add in `throw new RSError("test")` inside any of the network call hooks. Reload the page in the browser
5. You should see a generic error UI

Play around with sending the following errors on that line:
```typescript
RSNetworkError("You errored out!", 404)
RSNetworkError("You errored out!", 999)
Error("Test")
```

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #6522 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #6734 
- #6735

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
